pkg/controller/mpi_job_controller_test.go

```diff
@@ -151,7 +139,22 @@ func newMPIJobCommon(name string, startTime, completionTime *metav1.Time) *kubef
 
 func newMPIJob(name string, replicas *int32, startTime, completionTime *metav1.Time) *kubeflow.MPIJob {
 	mpiJob := newMPIJobCommon(name, startTime, completionTime)
-	mpiJob.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeWorker].Replicas = replicas
+	if *replicas > 0 {
```

Comment:
```suggestion
	if ptr.Deref(replicas, 0) > 0 {
```

It should avoid potentially null pointer.